lib/gds_api/middleware/govuk_header_sniffer.rb

```diff
@@ -0,0 +1,21 @@
+require_relative '../govuk_headers'
+
+module GdsApi
+  class GovukHeaderSniffer
+    def initialize(app, header_name)
+      @app = app
+      @header_name = header_name
+    end
+
+    def call(env)
+      GdsApi::GovukHeaders.set_header(readable_name, env[@header_name])
+      @app.call(env)
+    end
+
+    private
+
+    def readable_name
+      @header_name.sub(/^HTTP_/, "").gsub(/_/, "-").downcase.to_sym
```

Comment:
Actually, maybe it is unwanted... We want `"HTTP_GOVUK_REQUEST_ID"` to be transformed to `:govuk_request_id`, no?